src/simulation.hpp

```diff
@@ -112,6 +112,11 @@ template <typename problem_t> class AMRSimulation : public amrex::AmrCore
 	amrex::Real abstolPoisson_ = 1.0e-10; // default
 	int doPoissonSolve_ = 0;	      // 1 == self-gravity enabled, 0 == disabled
 
+	amrex::Real densityFloor_ = 1.67e-40; // in g cm-3, default
```

Comment:
Problems may not use cgs units. I think to minimize any accidental bugs, the floors should be set to zero by default and the ceiling should be set to infinity by default. This will just mean that if the user wants a ceiling or floor, they will need to set it appropriately for their problem.